tests/sles4sap/publiccloud/azure_fence_agents_test.pm

```diff
@@ -0,0 +1,90 @@
+# SUSE's openQA tests
+#
+# Copyright SUSE LLC
+# SPDX-License-Identifier: FSFAP
+# Maintainer: QE-SAP <qe-sap@suse.de>
+# Summary: Test module for azure stonith based fencing agent
+# Settings:
+#   FENCING_MECHANISM - needs to be set  to 'native'
+#   PUBLIC_CLOUD_PROVIDER - needs to be set to 'AZURE'
+#   AZURE_FENCE_AGENT_CONFIGURATION - set to 'msi' or 'spn' (default value: msi)
+#   SPN related settings:
+#       _SECRET_AZURE_SPN_APPLICATION_ID - application ID for fencing agent
+#       _SECRET_AZURE_SPN_APP_PASSWORD - application password used by fencing agent
+
+use strict;
+use warnings FATAL => 'all';
+
+use base 'sles4sap_publiccloud_basetest';
+use serial_terminal 'select_serial_terminal';
+use testapi;
+use sles4sap_publiccloud;
+use qesapdeployment;
+use Data::Dumper;
+
+sub test_flags {
+    return {fatal => 1, publiccloud_multi_module => 1};
+}
+
+sub run {
+    my ($self, $run_args) = @_;
+    my $instances = $self->{instances} = $run_args->{instances};
+    my $provider_client = $run_args->{instances}[0]{provider}{provider_client};
+    my $fence_agent_configuration = get_var('AZURE_FENCE_AGENT_CONFIGURATION', 'msi');
+    my $resource_group = qesap_az_get_resource_group();
+    my $subscription_id = $provider_client->{subscription};
+    my $tenant_id = check_var('AZURE_FENCE_AGENT_CONFIGURATION', 'spn') ? qesap_az_get_tenant_id($subscription_id) : '';
+    my $spn_application_id = get_var('_SECRET_AZURE_SPN_APPLICATION_ID');
+    my $spn_application_password = get_var('_SECRET_AZURE_SPN_APP_PASSWORD');
+    my @cluster_nodes = @{$self->list_cluster_nodes()};
+
+    die 'Resoruce group not found' unless $resource_group;
+    die 'Tenant ID is required in case of Azure SPN fencing' if check_var('AZURE_FENCE_AGENT_CONFIGURATION', 'spn') and !$tenant_id;
+    # Setting up credentials as variables so they do not show up in openQA outputs in plaintext
+    my $bashrc_vars = "export SUBSCRIPTION_ID=$subscription_id
```

Comment:
Should we create a second script to un-set them?

Reply:
Technically we could... not sure if it is a problem to keep em there. 
VM is eventually destroyed.
Btw those are in bashrc. Cannot set them via command because the output would be visible.